example/HelloWorldFunction/src/main/java/helloworld/AppIdempotency.java

```diff
@@ -0,0 +1,85 @@
+package helloworld;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.lambda.powertools.idempotency.Idempotency;
+import software.amazon.lambda.powertools.idempotency.IdempotencyConfig;
+import software.amazon.lambda.powertools.idempotency.Idempotent;
+import software.amazon.lambda.powertools.idempotency.persistence.DynamoDBPersistenceStore;
+import software.amazon.lambda.powertools.utilities.JsonConfig;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AppIdempotency implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private final static Logger LOG = LogManager.getLogger();
+
+    public AppIdempotency() {
+        // we need to initialize idempotency configuration before the handleRequest method is called
+        Idempotency.config().withConfig(
+                        IdempotencyConfig.builder()
+                                .withEventKeyJMESPath("powertools_json(body).address")
+                                .withUseLocalCache()
+                                .build())
+                .withPersistenceStore(
+                        DynamoDBPersistenceStore.builder()
+                                .withTableName("idempotency_table")
+                                .build()
+                ).configure();
```

Comment:
I know you've added javadocs everywhere with examples and in the exception, but is there anyway we can remove the need to call configure?

Reply:
We can probably remove it but it acts as the build() method of a builder. 

if you look at the code [here](https://github.com/jeromevdl/aws-lambda-powertools-java/blob/03b7b6c0edd75e5e408a865ae3cba87b67d5a9bc/powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/Idempotency.java#L85) you'll see that I perform a check (the persistence is mandatory) and create a default config if not set by the developer.
WDYT?

Reply:
👍🏻